src/main/scala/ai/starlake/job/ingest/XmlIngestionJob.scala

```diff
@@ -97,23 +94,23 @@ class XmlIngestionJob(
     * @param dataset
     *   input dataset as a RDD of string
     */
-  protected def ingest(dataset: DataFrame): (RDD[_], RDD[_]) = {
+  protected def ingest(dataset: DataFrame): (Dataset[String], Dataset[Row]) = {
+    import session.implicits._
     val datasetSchema = dataset.schema
     val errorList = compareTypes(schemaSparkType, datasetSchema)
-    val rejectedRDD = session.sparkContext.parallelize(errorList)
+    val rejectedDS = errorList.toDS
     metadata.xml.flatMap(_.get("skipValidation")) match {
       case Some(_) =>
-        val rejectedRDD = session.sparkContext.parallelize(errorList)
-        saveRejected(rejectedRDD, session.emptyDataFrame.rdd.map(_.mkString))
+        val rejectedDS = errorList.toDS()
+        saveRejected(rejectedDS, session.emptyDataset[String])
         saveAccepted(
-          dataset,
           ValidationResult(
-            session.sparkContext.emptyRDD[String],
-            session.sparkContext.emptyRDD[String],
-            session.sparkContext.emptyRDD[Row]
+            session.emptyDataset[String],
+            session.emptyDataset[String],
+            dataset
```

Comment:
we had `emptyRDD[Row]` now we have `dataset`
Are we sure we're not unintentionally changing behaviour? 
If we're fixing some misbehaving thing, are we sure we now have a unit test to cover this?

Reply:
oh no, saw the context just before this hunk. Never mind.